### ISSUE TITLE
AVIF  decode time  is   too  long

### ISSUE THREAD

**Issue by wangxuyang518** [bumptech/glide#4803]
avif decoder  is dependent on cpu  ，  sometimes  this decode process is  too long  、sush as spending  10s  or  5s  

Is there a good optimization scheme？

**Comment by vigneshvg** [comment]
Yes AVIF decoder is dependent on the CPU. But ideally it should not take that long on any modern device.

Do you have a sample image that takes 5-10s to load? If so, i can try and take a look at what's going on. Also, which device are you trying this on?

**Comment by wangxuyang518** [comment]
no explicit sample   ，decoding long time is a random behavior . 
we also consider whether the avif file size affects the decoding.   but  at present, no rules have been found

**Comment by vignesh-em** [comment]
[Android 12 introduces support for AVIF.](https://developer.android.com/about/versions/12/features#avif) 
Does this mean Glide can now take advantage of hardware acceleration (if available) to decode AVIFs?

**Comment by vigneshvg** [comment]
> [Android 12 introduces support for AVIF.](https://developer.android.com/about/versions/12/features#avif) Does this mean Glide can now take advantage of hardware acceleration (if available) to decode AVIFs?

Glide's AVIF integration is software only. So if you use that decoding will happen in software.

If you are using Glide without the AVIF integration, then AVIF images on Android 12+ will be decoded using the Android platform's decoder (which may be software or hardware depending on what the platform does). As far as i am aware, the platform does not use hardware AV1 decoder for AVIF decoding and uses software only. It is controlled by a [system property](https://cs.android.com/android/platform/superproject/+/master:frameworks/av/media/libmediaplayerservice/StagefrightMetadataRetriever.cpp;l=253-254;drc=688cec6e7adb88f3a121ab94d5aaa9ecd9a0d312) which is not set by default on devices that i checked (Pixel 6, Pixel 7).
